text/0000-cfg-logical-ops.md

```diff
@@ -0,0 +1,146 @@
+| `((((((a))))))`                    | a                                                  | `()` can be nested                                                                  |
+| `a && b \|\| c && d`               | `any(all(a, b), all(c, d))`                        | `\|\|` has lower precedence than `&&`                                               |
+| `a \|\| b && c \|\| d`             | `any(a, all(b, c), d)`                             | `\|\|` has lower precedence than `&&`                                               |
+| `(a \|\| b) && (c \|\| d)`         | `all(any(a, b), any(c, d))`                        | `()` can be used for grouping                                                       |
+| `!a \|\| !b && !c`                 | `any(not(a), all(not(b), not(c)))`                 | `!` has highest precedence                                                          |
+| `feature="foo" \|\| feature="bar"` | `any(feature="foo", feature="bar")`                | `\|\|` has lower precedence than `=`                                                |
+| `feature="foo" && feature="bar"`   | `all(feature="foo", feature="bar")`                | `&&` has lower precedence than `=`                                                  |
+| `!feature="foo"`                   | _syntax error_                                     | `!` has higher precedence than `=`, which may be confusing, so we ban this syntax   |
+| `!(feature="foo")`                 | `not(feature="foo")`                               | use `()` for grouping                                                               |
+| `!all(x, y)`                       | `not(all(x, y))`                                   | `!` has lower precedence than "function call"                                       |
+| `any(!x \|\| !w, !(y && z))`       | `any(any(not(x), not(w)), not(all(y, z)))`         | `!`, `&&` etc. can be used inside `any`, `all` and `not`                            |
+| `true && !false`                   | `all(true, not(false))`                            | `!`, `&&` etc. can be used on boolean literals (they are syntactically identifiers) |
+| `!accessible(std::mem::forget)`    | `not(accessible(std::mem::forget))`                | `!`, `&&` etc. can be used on `cfg_accessible`                                      |
+| `accessible(std::a \|\| std::b)`   | _syntax error_                                     | … but not inside                                                                    |
+| `!version("1.42.0")`               | `not(version("1.42.0"))`                           | `!`, `&&` etc. can be used on `cfg_version`                                         |
+| `version(!"1.42.0")`               | _syntax error_                                     | … but not inside                                                                    |
```

Comment:
How would it interact with the `target(os = "linux", ...)` shorthand from [RFC 3239](https://github.com/rust-lang/rfcs/pull/3239)?

Reply:
well it's being removed rust-lang/rust#130780

```rust
#[cfg(all(target_os = "linux", target_arch = "arm"))]
#[cfg(target_os("linux") && target_arch("arm"))]
#[cfg(target(os = "linux", arch = "arm"))]
```

Reply:
Last comments from T-lang members tend to the opposite, so probably going to stay.